labs/query_enriched_event_data_with_spark/index.md

```diff
@@ -0,0 +1,328 @@
+# Working with Enriched Events data using Spark SQL
+
+In this Lab we'll learn how to work with [Enriched Event Data](https://docs.developers.optimizely.com/web/docs/enriched-events-export) using [PySpark](http://spark.apache.org/docs/latest/api/python/index.html) and [Spark SQL](http://spark.apache.org/sql/).
+
+This lab contains a set of simple, useful queries for working with this dataset.  These queries can help you answer questions like
+- How many visitors were tested in my experiment?
+- How many "unique conversions" of an event were attributed to this variation?
+- What is the total revenue attributed to this variation?
+
+This Lab covers some of the basics of working with event-level experiment data. There are many more useful questions may want to answer with experiment data.  Future tutorials will go deeper on the topic.
+
+This guide borrows some initialization code from the [Spark SQL getting started guide](https://spark.apache.org/docs/latest/sql-getting-started.html).
+
+## Creating a Spark Session
+
+
+```python
+from pyspark.sql import SparkSession
+
+spark = SparkSession \
+    .builder \
+    .appName("Python Spark SQL") \
+    .config("spark.sql.repl.eagerEval.enabled", True) \
+    .getOrCreate()
+```
+
+## Loading enriched event data
+
+We'll start by loading decision and conversion data from `/data`.
+
+
+```python
+spark.read.parquet("data/decisions").createOrReplaceTempView("decisions")
+spark.read.parquet("data/events").createOrReplaceTempView("events")
+```
+
+## Querying our data
+
+Now that we've loaded our data, we can query it using the `sql()` function.  Here's an example on our `decisions` view:
+
+
+```python
+spark.sql("""
+    SELECT
+        *
+    FROM decisions
+    LIMIT 5
+""")
+```
+
+
+
+
+<table border='1'>
```

Comment:
If we stick with markdown, it might be easier to read this table if it was in the markdown table format. Or at least indented like a typical html file.

Reply:
Makes sense, but this is auto-generated by `jupyter nbconvert`.  I'll see if that tool provides any useful customization here.

Reply:
It doesn't look like there's much I can do about this, short of manually editing `index.md`, which i'd prefer not to do.